test/integration_test/health_monitor_test.go

```diff
@@ -61,3 +64,48 @@ func stopDriverTest(t *testing.T) {
 
 	destroyAndWait(t, ctxs)
 }
+
+func stopKubeletTest(t *testing.T) {
+	// Cordon node where the test is running. This is so that we don't end up stopping
+	// kubelet on the node where the stork-test pod is running
+	testPodNode := ""
+	testPod, err := k8s.Instance().GetPodByName("stork-test", "kube-system")
+	if err == nil { // if this hits an error, skip below logic to allow running tests outside a pod
+		testPodNode = testPod.Spec.NodeName
+		err = k8s.Instance().CordonNode(testPodNode, defaultWaitTimeout, defaultWaitInterval)
+		require.NoError(t, err, "Error cordorning k8s node for stork test pod")
+	}
+
+	ctxs, err := schedulerDriver.Schedule(generateInstanceID(t, "stopkubelettest"),
+		scheduler.ScheduleOptions{AppKeys: []string{"mysql-ss"}})
+	require.NoError(t, err, "Error scheduling task")
+	require.Equal(t, 1, len(ctxs), "Only one task should have started")
+
+	err = schedulerDriver.WaitForRunning(ctxs[0], defaultWaitTimeout, defaultWaitInterval)
+	require.NoError(t, err, "Error waiting for pod to get to running state")
+
+	scheduledNodes, err := schedulerDriver.GetNodesForApp(ctxs[0])
+	require.NoError(t, err, "Error getting node for app")
+	require.Equal(t, 1, len(scheduledNodes), "App should be scheduled on one node")
+
+	scheduledNode := scheduledNodes[0]
+	err = schedulerDriver.StopSchedOnNode(scheduledNode)
+	require.NoError(t, err, fmt.Sprintf("failed to stop scheduler on node: %s", scheduledNode.Name))
+
+	// wait for the scheduler daemon on node to stop and pod to get into unknown state
+	time.Sleep(6 * time.Minute)
+
+	err = schedulerDriver.WaitForRunning(ctxs[0], defaultWaitTimeout, defaultWaitInterval)
+	require.NoError(t, err, "Error waiting for pod to get to running state")
+
+	// restore scheduler
+	err = schedulerDriver.StartSchedOnNode(scheduledNode)
```

Comment:
This should be done in a defer so that failure in this test doesn't cause the following tests to fail.

Reply:
done